packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -174,7 +171,37 @@ class ChangeNotifier implements Listenable {
   @override
   void addListener(VoidCallback listener) {
     assert(_debugAssertNotDisposed());
-    _listeners!.add(_ListenerEntry(listener));
+    if (_length == _listeners.length) {
+      if (_length == 0) {
+        _listeners = List<VoidCallback?>.filled(1, null);
+      } else {
+        final List<VoidCallback?> newListeners =
+            List<VoidCallback?>.filled(_listeners.length * 2, null);
+        for (int i = 0; i < _length; i++) {
+          newListeners[i] = _listeners[i];
+        }
+        _listeners = newListeners;
+      }
+    }
+    _listeners[_length++] = listener;
+  }
+
+  void _removeAt(int index) {
+    _length--;
```

Comment:
style nit: we prefer `-= 1` to `--` for various reasons. See https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-over

Reply:
There are other occurrences of ++/-- in the code that still need updating.